src/notebook/tracker.ts

```diff
@@ -10,21 +10,21 @@ import {
 } from 'phosphor/lib/ui/focustracker';
 
 import {
-  NotebookPanel
-} from './';
+  Widget
+} from 'phosphor/lib/ui/widget';
 
 
 /* tslint:disable */
 /**
  * The notebook tracker token.
  */
 export
-const INotebookTracker = new Token<INotebookTracker>('jupyter.services.notebook-handler');
+const INotebookTracker = new Token<INotebookTracker>('jupyter.services.notebooks');
 /* tslint:enable */
 
 
 /**
  * A class that tracks notebook widgets.
  */
 export
-interface INotebookTracker extends FocusTracker<NotebookPanel> {}
+interface INotebookTracker extends FocusTracker<Widget> {}
```

Comment:
I rethought this as well, and now instead of giving back an instance of a `FocusTracker`, I simply give a `Map` with widget IDs that refer to widget instances. The only relevant focus tracking is done by the application shell now.